tensorflow/compiler/tf2tensorrt/convert/ops/fill_ops.cc

```diff
@@ -132,58 +132,72 @@ class ConvertRange : public ConvertFillBase<ConvertRange> {
     const auto& inputs = params.inputs;
     const auto& node_def = params.node_def;
 
-    if (!all_same_types(inputs)) {
-      return errors::InvalidArgument(convert_range_expected_msg(node_def),
-                                     "passed as weights OR tensors");
+    float param[3];
+    all_weights_ = all_integers_ = true;
+    for (int i = 0; i < 3; i++) {
+      const auto& input = inputs.at(i);
+      all_integers_ &= input.TrtDType() == nvinfer1::DataType::kINT32;
+      if (input.is_weights()) {
+        switch (input.TrtDType()) {
+          case nvinfer1::DataType::kFLOAT:
+            param[i] = get_input_param<float>(input);
+            break;
+          case nvinfer1::DataType::kHALF:
+            param[i] = get_input_param<Eigen::half>(input);
+            break;
+          default:  // nvinfer1::DataType::kINT32:
+            param[i] = get_input_param<int>(input);
+        }
+      } else {
+        all_weights_ = false;
+      }
     }
 
-    if (!all_weights_) {
-      if (!all_integers(inputs)) {
-        return errors::Unimplemented(convert_range_expected_msg(node_def),
-                                     "tensors");
+    if (!(all_weights_ || all_integers_)) {
+      // As of 06/03/2022, when at least one of the (start, limit, delta)
+      // is passed as a tensor, they must all be of type kINT32
+      return errors::Unimplemented(convert_range_expected_msg(node_def));
+    }
+
+    if (inputs.at(2).is_weights()) {
+      if ((delta_ = param[2]) == 0) {
+        return errors::InvalidArgument("The delta parameter of ", node_def.op(),
+                                       " operation cannot be equal to 0");
       }
 
-      for (int i = 0; i < 3; i++) {
-        const auto& dims = inputs.at(i).GetTrtDims();
-        if (dims.nbDims != 1 || dims.d[0] != 1) {
-          return errors::InvalidArgument("Dimension for '", InputSpec()[i].name,
-                                         "' of ", node_def.op(), " operator ",
-                                         "should be equal to 1");
-        }
+      if (!all_weights_ && delta_ < 0) {
```

Comment:
Why do we have this limitation?

Reply:
I use this limitation because, when `start` OR `limit` is passed as a tensor AND `delta` < 0, we get the following error messages:
```
2022-06-24 20:13:26.606085: I ./tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.h:96] Setting shape values for delta, nvinfer1::Dims(nbDims=1, d=-2,)
2022-06-24 20:13:26.606092: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:40] DefaultLogger 4: Specified optimization profiles must satisfy MIN<=OPT<=MAX.
2022-06-24 20:13:26.606220: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:40] DefaultLogger 3: [optimizationProfile.cpp::setShapeValues::196] Error Code 3: API Usage Error (Parameter check failed at: runtime/common/optimizationProfile.cpp::setShapeValues::196, condition: validate(mErrorStreams.getErrorRecorder(), newEntry, true)
)
2022-06-24 20:13:26.606230: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:40] DefaultLogger 4: Specified optimization profiles must satisfy MIN<=OPT<=MAX.
2022-06-24 20:13:26.606242: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:40] DefaultLogger 3: [optimizationProfile.cpp::setShapeValues::196] Error Code 3: API Usage Error (Parameter check failed at: runtime/common/optimizationProfile.cpp::setShapeValues::196, condition: validate(mErrorStreams.getErrorRecorder(), newEntry, true)
)
2022-06-24 20:13:26.606246: I ./tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.h:105] Setting input dimensions for delta, nvinfer1::Dims(nbDims=1, d=1,)
2022-06-24 20:13:26.606253: E tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.cc:388] Failed to add optimization profile [min: [nvinfer1::Dims(nbDims=1, d=1,),nvinfer1::Dims(nbDims=1, d=1,)], opt: : [nvinfer1::Dims(nbDims=1, d=1,),nvinfer1::Dims(nbDims=1, d=-2,)], max: [nvinfer1::Dims(nbDims=1, d=1,),nvinfer1::Dims(nbDims=1, d=-2,)]]. This usually happens when profile is invalid.
```